linera-core/src/updater.rs

```diff
@@ -296,28 +296,18 @@ where
                     let mut chain_heights = BTreeMap::new();
                     for certificate in certificates {
                         let block_chain_id = certificate.value().chain_id();
-                        let block_height = certificate.value().height();
+                        let block_height = certificate.value().height().try_add_one()?;
```

Comment:
This is changing the logic - is it still correct?

Reply:
There was a `try_add_one` in line 309 before. That's now gone, so we have to add one here, if I understand correctly.

Reply:
Yes, what @afck said. I just did the height increment earlier so we can call `send_chain_information_to_chain_heights` and reuse the code

Reply:
We might modify the entry in `chain_heights` prematurely though – previously the `block_height.max(*h)` might not modify `chain_heights` b/c the previously-stored `h` was higher. Now we'd compare it on the `block_height + 1`.